models/cube.js

```diff
@@ -126,7 +128,14 @@ const cubeSchema = mongoose.Schema({
         multiples: Boolean,
         html: String,
         markdown: String,
-        packs: String,
+        packs: [
+          {
+            filters: [String],
+            trash: { type: Number, min: 0 },
+            sealed: Boolean,
```

Comment:
sealed doesn't make sense as a pack variable, should be as a draft format property.

Also having picksPerPass and trash feels like a very weak way to customize each pack. I think we should instead have a 'steps' definition that looks something like (example is supreme draft):
`
steps: [
   {
      type: pick,
      count: 2,
   },
   {
       type: trash,
       count: 13,
   }
]
`

This would be more flexible and allow things like only the first pick is double. We would need to add validation to make sure that the numbers match up, but that wouldn't be too difficult.

Reply:
I think per pack sealed is fine, where the sealed format is just all packs sealed. Pack specificity allows for format flexibility (for example a commander draft where you open a sealed pack of commanders before the rest of the packs).

That being said, `{type: pick, count: N}` where N is the number cards in the pack is equivalent to sealed and even more generic. In either case we likely need special UI handling when sealed or picks == number of cards in pack (either an add all cards button or skip the step and add all cards to pool automatically).

In general I advocate for generic backend, but premade templates (e.g. supreme draft checkbox, etc) for frontend UI.

Reply:
I started implementing steps off the discord discussion before seeing this so the schema is
```js
{
                  action: {
                    type: String,
                    enum: ['pass', 'pick', 'trash', 'pickrandom', 'trashrandom'],
                  },
                  amount: {
                    type: Number,
                    default: 1,
                  },
                },
```
Amount defaults to 1 and is ignored for pass (open to changing this to allow for really esoteric formats with positive and negative allowing for customizing direction. It would require very little change to the code). 
I intend to add a feature that gives you a pick/trash all button for when the trash or pick amount is equal to the rest of the pack.

I do think UI shortcuts to make defaults would be good. I'm thinking that per pack you could autofill: Supreme, Sealed, Pick N at a time, Trash last N, 2XM, and maybe more if we get requests.